pegasus-spark-bulkloader/src/main/java/com/xiaomi/infra/pegasus/spark/bulkloader/DataWriter.java

```diff
@@ -0,0 +1,62 @@
+package com.xiaomi.infra.pegasus.spark.bulkloader;
+
+import com.github.rholder.retry.RetryException;
+import com.github.rholder.retry.Retryer;
+import com.xiaomi.infra.pegasus.spark.PegasusSparkException;
+import com.xiaomi.infra.pegasus.spark.RocksDBOptions;
+import com.xiaomi.infra.pegasus.spark.Tools;
+import java.util.concurrent.ExecutionException;
+import org.rocksdb.RocksDBException;
+import org.rocksdb.SstFileWriter;
+
+/** The wrapper of sstFileWriter of rocksdbjava */
+class DataWriter {
+
+  private SstFileWriter sstFileWriter;
+  private Retryer<Boolean> booleanRetryer = Tools.getDefaultRetryer();
+  private Retryer<Integer> integerRetryer = Tools.getDefaultRetryer();
+
+  DataWriter(RocksDBOptions rocksDBOptions) {
+    this.sstFileWriter = new SstFileWriter(rocksDBOptions.envOptions, rocksDBOptions.options);
+  }
+
+  void openWithRetry(String path) throws PegasusSparkException {
+    try {
+      booleanRetryer.call(() -> open(path));
+    } catch (ExecutionException | RetryException e) {
+      throw new PegasusSparkException("sstFileWriter open [" + path + "] failed!", e);
+    }
+  }
+
+  private boolean open(String path) throws RocksDBException {
+    sstFileWriter.open(path);
+    return true;
+  }
+
+  int writeWithRetry(byte[] key, byte[] value) throws PegasusSparkException {
+    try {
+      return integerRetryer.call(() -> write(key, value));
+    } catch (ExecutionException | RetryException e) {
+      throw new PegasusSparkException(
+          "sstFileWriter put key-value[key=" + new String(key) + "] failed!", e);
+    }
+  }
+
+  private int write(byte[] key, byte[] value) throws RocksDBException {
+    sstFileWriter.put(key, value);
+    return key.length + value.length;
+  }
+
+  void closeWithRetry() throws PegasusSparkException {
+    try {
+      booleanRetryer.call(this::close);
+    } catch (ExecutionException | RetryException e) {
+      throw new PegasusSparkException("sstFileWriter close failed!", e);
+    }
+  }
+
+  public boolean close() throws RocksDBException {
```

Comment:
```suggestion
  private boolean close() throws RocksDBException {
```

Should this be a `private`?

Reply:
Shouldn't, it's used by https://github.com/pegasus-kv/pegasus-spark/blob/6dcaeb327a422a77713a8c259fd113b716e0f0d7/pegasus-spark-analyser/src/main/java/com/xiaomi/infra/pegasus/spark/analyser/ColdBackupLoader.java#L209-L212